release_notes/v2.2.0.md

```diff
@@ -4,23 +4,166 @@ v2.2.0 Release Notes - July XX, 2020
+with the HSM.  It was possible for the fatal log to occur when an expected error
+occurred, i.e., the HSM had reached its maximum session handles.
+Instead of issuing a fatal log call, an error is returned so that
+the error can be properly handled.
+
+**FAB-17819: Discovery returns user friendly errors**
+
+Service discovery endorsement service error message "cannot satisfy any principal combination" is
+improved to return a more specific message, either "no peer combination can satisfy the endorsement policy"
+or "required chaincodes are not installed on sufficient peers".
+
+**FAB-17774: Support orderer restart without system channel genesis block**
+
+When BootstrapMethod was set to 'file', the system channel genesis block
+was required to be passed for every orderer service start. The system
+channel genesis block is now only required for the initial orderer start.
+
+**FAB-17844: External builder fails to copy symlinks from build output into persistent directory**
+
+Previously, the external builder code did not check for symlinks in build output when copying them.
+This resulted in the resolved files being copied as files instead of symlinks. The external builder
+now copies them as symlinks instead of copying them as files into the destination directory.
+
+**Errors should be checked when orderer grpc server is serving requests**
+
+grpc errors are now checked when servicing the orderer atomic broadcast grpc service.
+
+**FAB-17900: Fix environment variable override bug**
+
+Integer config values for peer and orderer could not be overridden with environment variables.
+
+**FAB-17951: Fetch correct node id for orderer consenter validation**
+
+Config update validator may incorrectly reject updates if some nodes are inactive.
+
+**FAB-17875: Fix ordering service node leader election failure**
+
+Previously with raft consensus, when one ordering service node was deleted
+from a channel and rejoined later, it would be assigned a new raft id.
+However, in some cases the ordering service node still used the old raft
+id. Other ordering service nodes including the leader are using the latest view and believe
+that the rejoined node would use a new raft id.  This may result in leader election failure and no new
+transactions would be accepted. The fix ensures the correct raft id is used after tracking the latest config block.
+
+**Private data performance optimization: purge transient store in background**
+
+Private data performance is improved by purging entries present in the transient store in the background.
+
+**FAB-17933: Fix cache update logic for installed chaincode info when an empty or uninstalled package ID is specified**
+
+Previous cache update logic for install chaincode info did not work properly when an empty or uninstalled package ID was specified.
+
+**FAB-17539: Always remember gossip anchor peers in membership**
+
+Previously gossip service removed a peer from its membership cache if the peer's alive message was expired.
```

Comment:
Gossip removes a peer from its membership cache if no new heartbeats are received from the peer within a timely manner.

If a network partition persists for too long, peers of different organizations never re-establish communication because all membership is purged.

Now, anchor peers are no longer removed from the membership cache even if they are offline.

Therefore, after the network partition is healed, peers among different organizations can reestablish communication as long as anchor peers are reachable.

Reply:
Thanks, updated.